move waived param to yaml

wizardofoz123 opened the pull request:
This requires complementary kpet-db change.

Signed-off-by: Jakub Racek <jracek@redhat.com>

spbnick commented on kpet/data.py:
```diff
@@ -283,6 +288,7 @@ def convert(old_data):
                         kickstart=String(),
                         match=Class(PositivePattern),
                         dont_match=Class(NegativePattern),
+                        fetch_suffix=String(),
```
Could you please rename this to `url_suffix`? That would be clearer, I think.

spbnick commented on kpet/data.py:
```diff
@@ -266,13 +269,15 @@ def convert(old_data):
                         description=String(),
-                        cases=List(Class(Case))
+                        cases=List(Class(Case)),
+                        maintainer=String(),
```
Some tests have multiple maintainers, so could you please name it `maintainers` and make it a list of strings instead? Also, please make it optional with an empty list as the default (replacing `None` with `[]` right here in the constructor). This way we also won't have to update so many test assets.

wizardofoz123 replied:
I keep thinking that every test should have a point of contact (some maintainer), never no-one. Wasn't this one of the requirements for onboarding?

spbnick replied:
Weeell, to be honest, I'm not really sure about that. I think it's a good idea to make it required. However, we would need to add support for validating lists with a minimum number of elements (to be able to have a maintainer list and check that there's at least one). If you could do that, then I'm game :)

spbnick commented on kpet/data.py:
```diff
@@ -266,13 +269,15 @@ def convert(old_data):
                         hostRequires=String(),
                         partitions=String(),
                         kickstart=String(),
+                        fetch_suffix=String(),
```
Please don't update the legacy schema. First, we shouldn't change the past, and then we need to remove support for it (and update the tests/assets) at the first opportunity, since the database has migrated away.

spbnick commented on tests/assets/db/general/trees/rhel7.xml:
```diff
@@ -1,15 +1,39 @@
-        {% include case[template_field] %}
-      {% endif %}
+        {% if template_field == 'tasks' %}
+            {% if case['role'] is not none %}
```
Casserole :shallow_pan_of_food:, yummy :grin:!

wizardofoz123 replied:
I gather that you don't actually need any changes here. (And yes, the naming is funny, but that was purely by accident.)

spbnick replied:
Exactly, just sharing the amusing coincidence :smile:.

spbnick commented on tests/assets/db/general/trees/rhel7.xml:
```diff
@@ -1,15 +1,39 @@
+            {% if case[template_field] %}
+                {% include case[template_field] %}
+            {% endif %}
+        {% endif %}
```
I'll leave some comments about this part in the kpet-db change. Please update here as well.

spbnick commented on tests/assets/db/general/trees/rhel7.xml:
```diff
@@ -55,10 +79,10 @@
         </task>
         <task name="/test/misc/machineinfo">
-          <fetch url="https://github.com/CKI-project/tests-beaker/archive/staging.zip#test/misc/machineinfo"/>
+          <fetch url="https://github.com/CKI-project/tests-beaker/archive/master.zip#test/misc/machineinfo"/>
```
Please make the URL changes a separate commit.

spbnick commented on tests/assets/db/general/suites/default/index.yaml:
```diff
@@ -1,7 +1,5 @@
 cases:
-    -   name: default/ltplite
-        tasks: suites/default/ltplite.xml
+    -   name: /kernel/distribution/ltp/lite
```
Please make this case name change a separate commit.

wizardofoz123 replied:
The generated task name in XML doesn't actually change. The original has 
<task name="/kernel/distribution/ltp/lite" role="STANDALONE"> , so this change is here to make sure the output remains the same. Are you sure you want to split this?

spbnick replied:
Ah, if I got it right, and this is just bringing in the task's name, like all those changes in kpet-db, then sure, please keep it! Thanks :)

spbnick commented on kpet/data.py:
```diff
@@ -213,6 +213,8 @@ def __init__(self, data):
                     dont_match=Class(NegativePattern),
                     waived=Boolean(),
+                    role=String(),
+                    extra_params=String(),
```
Even though `role` is Beaker-specific, we'll need it as is for now. However, we have a good opportunity to remove Beaker-dependence and obscurity with `extra_params`. Let's just call it `environment`, or `env_vars`, or something like that. It will be clear, to the point, and non-Beaker specific. Finally make it `Dict(String())` to represent variable names/values directly, instead of having to paste XML into YAML, where we don't have to. Make the default be an empty dictionary too to simplify templates.

spbnick reviewed (changes requested):
Thank you, Jakub. This is going in the right direction, no question. However, I have a bunch of requests below, and inline.

Please keep pull request title and description up-to-date, here and on GitLab. It's difficult for me to recall which PR is which and what it does otherwise.

Please also keep logically-unrelated changes in separate commits (concerns kpet-db changes) and parts of the same logical change in a single commit. I.e. have "move waived param to yaml" and "move all params to yaml" in a single commit together. OTOH, have changing the tests-beaker URL, and case names in test assets separate from everything else and each other.

Also, please make sure every commit (not just every PR) works. In this case, the tests would stop working after the schema update, so the test change would need to be in the same commit with the schema change.

The above makes it easier for me to review/drop/merge/revert/think about specific commits.

spbnick commented on kpet/data.py:
```diff
@@ -213,6 +213,8 @@ def __init__(self, data):
                     match=Class(PositivePattern),
                     dont_match=Class(NegativePattern),
                     waived=Boolean(),
+                    role=String(),
```
Another thing: please assign the default `role` right here in the code, so you don't have to do it in the template and can just assume it always exists.

Commit:
remove link to staging

The XML we generate during tests will have <fetch > element adjusted by
url_prefix. It shouldn't really contain link to staging that we don't
use; this makes creating new assets/tests more complicated.

Signed-off-by: Jakub Racek <jracek@redhat.com>

Commit:
adjust all tests for suites tasks

Integration and unit tests for kpet have a lot of assets. Currently, to take
advantage of the pending kpet-db MR that gets rid of all task XML
snippets, 'maintainer: ' has to be added to the yamls. Field fetch_suffix
is not mandatory, but desirable.
Make maintainers non-empty optional list of strings.

The jinja template is adjusted to generate XML snippets instead of
loading them from files.

Changes over previous versions:
* overhaul completely
* pylint fixes
* add fixes requested in review
* changes should pass tests independently

Signed-off-by: Jakub Racek <jracek@redhat.com>
